File: Cpp/242_Valid_Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()) return false;
        
        map<char, int> count;
        int n = s.size();
        for(int i=0; i<n; ++i){
            ++count[s[i]];
            --count[t[i]];
        }
        
        for(auto& c : count){
            if(c.second != 0) return false;
        }
        
        return true;
    }
};

File: Kotlin/242_Valid_Anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        if(s.length != t.length) return false
        
        val count = IntArray(26)
        
        for(i in s.indices) {
            ++count[s[i] - 'a']
            --count[t[i] - 'a']
        }
        
        for(i in count) {
            if(i != 0) return false
        }
        
        return true
    }
}